.github/workflows/dotnet.yml

```diff
@@ -0,0 +1,39 @@
+name: Chorus .NET6
+
+on:
+  [push, pull_request, workflow_dispatch]
+
+jobs:
+  build_test_deploy:
+    env:
+      NUGET_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    permissions: write-all
+    
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # fetch full history for GitVersion
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+        source-url: https://nuget.pkg.github.com/sillsdev/index.json
+    - name: Fix Python on Linux (for mercurial)
+      if: matrix.os == 'ubuntu-latest'
+      run: sudo apt install python-is-python2
+    - name: Restore
+      run: dotnet restore
+    - name: Build
+      run: dotnet build --no-restore -c Release
+    - name: Test
+      run: dotnet test
+    - name: Pack
+      if: github.ref == 'refs/heads/master'
+      run: dotnet pack --no-restore --no-build -c Release
+    - name: Deploy
+      if: github.ref == 'refs/heads/master'
```

Comment:
We don't want to do the Deploy step on both Windows and Linux, and to be on the save side also add a check `github.event_name == 'push'` (I'm not entirely sure that's needed, but at least it makes it explicit that it should only happen when a new commit gets pushed to master and not on PRs).